libraries/chain/sps_objects/sps_processor.cpp

```diff
@@ -0,0 +1,286 @@
+   {
+      itr = sps_helper::remove_proposal< by_end_date >( itr, proposalIndex, votesIndex, byVoterIdx, obj_perf );
+      if( obj_perf.done )
+         break;
+   }
+}
+
+void sps_processor::find_active_proposals( const time_point_sec& head_time, t_proposals& proposals )
+{
+   const auto& pidx = db.get_index< proposal_index >().indices().get< by_start_date >();
+
+   std::for_each( pidx.begin(), pidx.upper_bound( head_time ), [&]( auto& proposal )
+                                             {
+                                                if( head_time >= proposal.start_date && head_time <= proposal.end_date )
+                                                   proposals.emplace_back( proposal );                                                
+                                             } );
+}
+
+uint64_t sps_processor::calculate_votes( const proposal_id_type& id )
+{
+   uint64_t ret = 0;
+
+   const auto& pvidx = db.get_index< proposal_vote_index >().indices().get< by_proposal_voter >();
+   auto found = pvidx.find( id );
+
+   while( found != pvidx.end() && found->proposal_id == id )
+   {
+      const auto& _voter = db.get_account( found->voter );
+
+      //If _voter has set proxy, then his votes aren't taken into consideration
+      if( _voter.proxy == STEEM_PROXY_TO_SELF_ACCOUNT )
+      {
+         auto sum = std::accumulate(   _voter.proxied_vsf_votes.begin(),
+                                       _voter.proxied_vsf_votes.end(),
+                                       _voter.vesting_shares.amount );
+         ret += sum.value;
+      }
+
+      ++found;
+   }
+
+   return ret;
+}
+
+void sps_processor::calculate_votes( const t_proposals& proposals )
+{
+   for( auto& item : proposals )
+   {
+      const proposal_object& _item = item;
+      auto total_votes = calculate_votes( _item.proposal_id );
+
+      db.modify( _item, [&]( auto& proposal )
+                        {
+                           proposal.total_votes = total_votes;
+                        } );
+   }
+}
+
+void sps_processor::sort_by_votes( t_proposals& proposals )
+{ 
+   std::sort( proposals.begin(), proposals.end(), []( const proposal_object& a, const proposal_object& b )
+                                                      {
+                                                         if (a.total_votes == b.total_votes)
+                                                         {
+                                                            return a.proposal_id < b.proposal_id;
+                                                         }
+                                                         return a.total_votes > b.total_votes;
+                                                      } );
+}
+
+asset sps_processor::get_treasury_fund()
+{
+   auto& treasury_account = db.get_account( STEEM_TREASURY_ACCOUNT );
+
+   return treasury_account.sbd_balance;
+}
+
+asset sps_processor::get_daily_inflation()
+{
+   //temporary
+   return asset( 0, SBD_SYMBOL );
```

Comment:
We should have the proper logic here, even though the inflation rate is 0 so that when it is changed later, this method does not get overlooked. At a minimum, this needs an FCTODO as a reminder.